Dockerfile

```diff
@@ -27,12 +28,12 @@ RUN git clone https://github.com/mendersoftware/mender.git . && \
     git checkout $MENDER_CLIENT_VERSION && \
     DESTDIR=/install-modules-gen make install-modules-gen
 
-FROM debian:12.1-slim
+FROM docker:24-dind
+ARG MENDER_APP_UPDATE_MODULE_VERSION
 COPY --from=cli-builder /go/src/github.com/mendersoftware/mender-cli /usr/bin/
 COPY --from=artifact-builder /go/src/github.com/mendersoftware/mender-artifact/mender-artifact /usr/bin/
 COPY --from=client-builder /install-modules-gen/usr/bin/ /usr/bin/
 # The --mount instead of COPY is used to decrease the image layer size. Requires: 'export DOCKER_BUILDKIT=1'
-RUN --mount=type=bind,source=deb-requirements.txt,target=/deb-requirements.txt \
-    apt-get update && \
-    apt-get install -y $(cat deb-requirements.txt) && \
-    rm -rf /var/lib/apt/lists/*
+RUN wget https://raw.githubusercontent.com/mendersoftware/app-update-module/$MENDER_APP_UPDATE_MODULE_VERSION/gen/app-gen -O /usr/bin/app-gen && chmod +x /usr/bin/app-gen
```

Comment:
in all the remaining cases we are cloning repos, more or less idle comment: shall we clone or shall we wget?

Reply:
don't know, let's wait for @lluiscampos and see what he thinks about it

Reply:
Both approaches are fine.

I favor the existing approach in that it encapsulates (part of) the complexity by using the `Makefile` as the interface. But one can also argue that it is unnecessary and that we anyway leak the paths for the executable, so doing `wget` directly on the source is not that off after all.